extension/history.js

```diff
@@ -0,0 +1,129 @@
+/* globals log */
+
+export class Database {
+  constructor(dbName) {
+    this.DBName = dbName;
+    this.readonly = "readonly";
+    this.readwrite = "readwrite";
+  }
+
+  createTable(TBName, primaryKey) {
+    return new Promise((resolve, reject) => {
+      const request = indexedDB.open(this.DBName);
+      request.onupgradeneeded = e => {
+        const database = e.target.result;
+        database.createObjectStore(TBName, {
+          keyPath: primaryKey,
+        });
+      };
+      request.onsuccess = e => {
+        const database = e.target.result;
+        resolve(database);
+      };
+      request.onerror = function(e) {
+        reject("Error: failed to create a database!");
+      };
+    });
+  }
+
+  get(primaryKey, TBName) {
+    return new Promise((resolve, reject) => {
+      const request = indexedDB.open(this.DBName);
+      request.onsuccess = e => {
+        const database = e.target.result;
+        const transaction = database.transaction([TBName]);
+        const objectStore = transaction.objectStore(TBName);
+        const read = objectStore.get(primaryKey);
+
+        read.onerror = function(event) {
+          reject("Unable to retrieve data from database!");
+        };
+
+        read.onsuccess = function(event) {
+          log.info("get result:", read.result);
+          resolve(read.result);
+        };
+        database.close();
+      };
+      request.onerror = event => reject("Unable to retrieve from database!");
+    });
+  }
+
+  getAll(TBName) {
```

Comment:
I believe this is the main method we'll want to be able to call, and we'll probably want to be able to get the items in sorted order (for us, sorted by timestamp, most recent first).